crates/oxc_linter/src/rules/eslint/no_label_var.rs

```diff
@@ -0,0 +1,69 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_label_var_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("eslint(no-label-var): Found identifier with same name as label.")
+        .with_label(span0)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoLabelVar;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow labels that share a name with a variable.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// This rule aims to create clearer code by disallowing the bad practice of creating a label
+    /// that shares a name with a variable that is in scope.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var x = foo;
+    /// function bar() {
+    /// x:
+    ///   for (;;) {
+    ///     break x;
+    ///   }
+    /// }
+    /// ```
+    NoLabelVar,
+    style,
+);
+
+impl Rule for NoLabelVar {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::LabeledStatement(labeled_stmt) = node.kind() else { return };
+
+        let scope = ctx.scopes();
+        if scope.ancestors(node.scope_id()).any(|v| {
+            scope.get_bindings(v).iter().any(|(k, _)| k.as_str() == labeled_stmt.label.name)
+        }) {
+            ctx.diagnostic(no_label_var_diagnostic(labeled_stmt.span));
```

Comment:
Consider adding label spans for the label and variable declaration nodes.

Reply:
I looked at what the eslint rule did, and it just selects the `full label declaration`. It would be nicer to also mark the declarations I'll look into doing that.